glean-core/android/src/main/java/mozilla/telemetry/glean/scheduler/GleanLifecycleObserver.kt

```diff
@@ -4,17 +4,30 @@
 
 package mozilla.telemetry.glean.scheduler
 
+import android.content.Context
+import android.content.SharedPreferences
 import androidx.lifecycle.Lifecycle
 import androidx.lifecycle.LifecycleEventObserver
 import androidx.lifecycle.LifecycleOwner
 import mozilla.telemetry.glean.Glean
 import mozilla.telemetry.glean.GleanMetrics.GleanBaseline
+import mozilla.telemetry.glean.GleanMetrics.GleanValidation
 
 /**
  * Connects process lifecycle events from Android to Glean's handleEvent
  * functionality (where the actual work of sending pings is done).
  */
-internal class GleanLifecycleObserver : LifecycleEventObserver {
+internal class GleanLifecycleObserver(
+    private val applicationContext: Context
+) : LifecycleEventObserver {
+    // Using SharedPreferences to track the status here should be less of an I/O hit
+    // compared to plain file usage, given that most of its operations are off the main
+    // thread. This also gives us the nice guarantee that once the function is called
+    // the OS will keep SharedPreferences working, even though the process is being killed.
```

Comment:
Not sure what this means. What does it mean to "keep SharedPreferences working"?

Reply:
I mean that `"You don't need to worry about Android component lifecycles and their interaction with apply() writing to disk. The framework makes sure in-flight disk writes from apply() complete before switching states. "`, from [the docs](https://developer.android.com/reference/android/content/SharedPreferences.Editor.html#apply()).